packages/suite-desktop/src-electron/modules/tor.ts

```diff
@@ -38,55 +38,81 @@ const init: Module = async ({ mainWindow, store, interceptor }) => {
         });
     };
 
+    const getProxySettings = (shouldEnableTor: boolean) =>
+        shouldEnableTor
+            ? {
+                  proxy: `socks://${address}`,
+                  useOnionLinks: true,
+              }
+            : { proxy: '', useOnionLinks: false };
+
     const setupTor = async (settings: TorSettings) => {
-        const shouldRunBundledTor = settings.running;
-        if (settings.running !== (await tor.status()).process) {
-            if (shouldRunBundledTor === true) {
-                try {
-                    await tor.start();
-                } catch (error) {
-                    logger.error('tor', `Failed to start: ${error.message}`);
-                    captureException(error);
-                }
-            } else {
-                try {
-                    await tor.stop();
-                } catch (error) {
-                    logger.error('tor', `Failed to stop: ${error.message}`);
-                    captureException(error);
-                }
-            }
+        const shouldEnableTor = settings.running;
+        const isTorRunning = (await tor.status()).process;
+
+        if (shouldEnableTor === isTorRunning) {
+            return;
         }
 
-        // Start (or stop) routing all communication through tor.
-        if (settings.running) {
+        if (shouldEnableTor === true) {
+            await tor.start();
+
             setProxy(`socks5://${host}:${port}`);
         } else {
+            await tor.stop();
             setProxy('');
         }
 
+        persistSettings(settings);
+
         // Notify the renderer.
         mainWindow.webContents.send('tor/status', settings.running);
```

Comment:
We are now getting info from electron layer twice.
<img width="783" alt="Screenshot 2022-06-21 at 10 25 12" src="https://user-images.githubusercontent.com/33235762/174753192-bd3551ee-bd91-4061-8d64-90aef0d60ccd.png">

Reply:
I suppose that should be solved in another branch?

Reply:
sure